src/app/services/date/date.service.ts

```diff
@@ -22,8 +22,14 @@ export class DateService {
 
 	// TODO: Implement agreegating apiDatas by: DAY, WEEK, MONTH
 	getAggregatedReigstryData(apiDatas: RegistryData[], dates: Date[]): (string | number)[][] {
+		// may happen that dates.length is more than apiDatas[0].range.length
+		const minimumRangeLength = apiDatas.reduce(
+			(acc, curr) => (acc > curr.range.length ? curr.range.length : acc),
+			1000000
+		);
+
 		// MM/dd vs MM/dd/yy
-		const rows = dates.map((date, i) => {
+		const rows = dates.slice(-minimumRangeLength).map((date, i) => {
 			return [format(date, 'MM/dd/yy'), ...apiDatas.map((apiData) => apiData.range[i].downloads)];
 		});
```

Comment:
Before using `slice` it used to throw an error “ERROR TypeError: Cannot read properties of undefined (reading 'downloads’)” because `dates` contained more data than the array length in any element in apiData (apiDatas[0].range.length)

Reply:
Is this resolved?